src/main/kotlin/com/varabyte/kobweb/intellij/colors/KobwebColorProvider.kt

```diff
@@ -0,0 +1,222 @@
+
+        is KtCallExpression -> (null).apply {
+            val callee = ((element.calleeExpression as? KtNameReferenceExpression)?.locateSource() as? KtNamedFunction)
+                ?: return@apply
+
+            when {
+                callee.isKobwebColorFunction("rgb(r: Int, g: Int, b: Int)") -> run {
+                    val args = element.valueArguments.evaluateArguments<Int>(3) ?: return@run
+                    return safeRgbColor(args[0], args[1], args[2])
+                }
+
+                callee.isKobwebColorFunction("rgb(value: Int)") -> run {
+                    val args = element.valueArguments.evaluateArguments<Int>(1) ?: return@run
+                    return safeRgbColor(args[0])
+                }
+
+                callee.isKobwebColorFunction("rgba(value: Int)", "rgba(value: Long)") -> run {
+                    val args = element.valueArguments.run {
+                        evaluateArguments<Int>(1) ?: evaluateArguments<Long, Int>(1) { it.toInt() }
+                    } ?: return@run
+                    return safeRgbColor(args[0] shr Byte.SIZE_BITS)
+                }
+
+                callee.isKobwebColorFunction("argb(value: Int)", "argb(value: Long)") -> run {
+                    val args = element.valueArguments.run {
+                        evaluateArguments<Int>(1) ?: evaluateArguments<Long, Int>(1) { it.toInt() }
+                    } ?: return@run
+                    return safeRgbColor(args[0] and 0x00_FF_FF_FF)
+                }
+
+                callee.isKobwebColorFunction("hsl(h: Float, s: Float, l: Float)") -> run {
+                    val args = element.valueArguments.evaluateArguments<Float>(3) ?: return@run
+                    return safeHslColor(args[0], args[1], args[2])
+                }
+
+                callee.isKobwebColorFunction("hsla(h: Float, s: Float, l: Float, a: Float)") -> run {
+                    val args = element.valueArguments.evaluateArguments<Float>(4) ?: return@run
+                    return safeHslColor(args[0], args[1], args[2])
+                }
+            }
+        }
+
+        else -> null
+    }
+
+    return if (currentDepth <= MAX_SEARCH_DEPTH) {
+        nextElement?.let { traceColor(it, currentDepth + 1) }
+    } else null
+}
+
+/**
+ * Pretty much emulates what CTRL-clicking on a reference would do
+ */
+private fun KtSimpleNameExpression.locateSource(): PsiElement? {
+    val resolvedMainReference = this.mainReference.resolve() ?: return null
+    val sourceDescriptor =
+        PsiNavigationSupport.getInstance().getDescriptor(resolvedMainReference) as? OpenFileDescriptor ?: return null
+    val sourcePsiFile = sourceDescriptor.file.toPsiFile(this.project) ?: return null
+    val sourceElementIdentifier = PsiUtilBase.getElementAtOffset(sourcePsiFile, sourceDescriptor.offset)
+    val sourceElement = sourceElementIdentifier.parent
+
+    return sourceElement
+}
+
+/**
+ * Uses the kotlin analysis api, as it can parse the constants much smarter.
```

Comment:
Note there are two broad types of comments: "header docs", and, uh, just "comments" :stuck_out_tongue: 

`/** ... */` is header doc comments and their purpose is to explain the "what" of a function, and other useful details. It's something that should help the user understand the code without having to read it.

At some point, please review https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html especially the principles section. See also https://kotlinlang.org/docs/kotlin-doc.html which layers on Kotlin-specific functionality.

`// ...` is a regular comment, and it should almost always only exist to explain "why", never "what". (This is actually not a mistake I see you making a lot so far which is good. When comments just repeat the code, they become noise, and worse, they can come out of date.

This comment here feels part regular comment and part header doc.

So first, I would recommend splitting it up, like this:

```
/**
 * Evaluates a collection of value arguments, returning some resolved value.
 *
 * For example, if we have a collection of decimal, hex, and binary arguments, this method can parse
 * them into raw values.
 */
// NOTE: We delicate the hard work to the Kotlin analysis API, as it provides robust parsing logic for us.
private inline fun ... {
```

Now that they're split up, we ask the next question... is the regular comment really necessary? Here, it has a little bit of "why" in it (that is, "the kotlin analysis API is better at it!"). However, that doesn't seem really helpful to me; it could be reworded at "We call that code instead of doing it ourselves because that code is good, and works already". Which should be all code everywhere! :)